app/src/main/java/com/cornellappdev/scoop/ui/login/AuthScreen.kt

```diff
@@ -0,0 +1,172 @@
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.Response
+import org.json.JSONObject
+import java.io.IOException
+
+
+@Composable
+fun AuthFlow(signInClient: GoogleSignInClient) {
+
+    val navController = rememberNavController()
+
+    // Set up the NavHost with the navigation graph
+    NavHost(navController = navController, startDestination = "AUTH") {
+        composable("AUTH") {
+            AuthScreen(signInClient, navController)
+        }
+        composable("HOME") {
+            MainScreen()
+
+        }
+    }
+
+}
+
+
+@Composable
+fun AuthScreen(signInClient: GoogleSignInClient, navController: NavHostController) {
+
+    val startForResult =
+        rememberLauncherForActivityResult(ActivityResultContracts.StartActivityForResult()) { result: ActivityResult ->
+            if (result.resultCode == Activity.RESULT_OK) {
+                val intent = result.data
+                if (result.data != null) {
+                    val task: Task<GoogleSignInAccount> =
+                        GoogleSignIn.getSignedInAccountFromIntent(intent)
+
+                    task.addOnCompleteListener { res ->
+                        if (res.isSuccessful) {
+                            val account = res.result
+                            // Successful sign in
+                            val res =
+                                SignedInUser(
+                                    firstName = account.givenName!!,
+                                    lastName = account.familyName!!, id = account.id!!,
+                                    idToken = account.idToken!!, email = account.email!!
+                                )
+
+                            LoginRepository.retrieveAccessToken(res).enqueue(object : Callback {
+
+                                override fun onFailure(call: Call, e: IOException) {
+                                    Log.d("Signin", "Sign in error")
+                                }
+
+                                override fun onResponse(call: Call, response: Response) {
+
+                                    // Parse response
+                                    val responseJson = JSONObject(response.body!!.string())
+
+                                    var accessToken = responseJson.getString("access_token")
+
+                                    response.close()
+                                    Log.d("Signin", "Successfully authenticated $accessToken")
+
+                                    LoginRepository.headers["Authorization"] = "Token $accessToken"
+
+                                    CoroutineScope(Dispatchers.Main).launch {
+                                        navController.navigate("HOME")
+                                    }
+
+                                }
+                            })
```

Comment:
I feel like this is code logic you might be able to move over to a custom function in `LoginRepository`. It's best to try to separate Model from View whenever possible to clean up our code & practice good abstraction.

It's mostly fine though if you can move that code over and then just call that function here.